src/o365/spfx/commands/project/project-upgrade/model/ConfigJson.ts

```diff
@@ -6,10 +6,19 @@ export interface ConfigJson {
   entries?: Entry[];
   localizedResources?: Hash;
   version?: string;
+  externals?:ExternalConfiguration;
 }
 
 export interface Entry {
   entry: string;
   manifest: string;
   outputPath: string;
+}
+export interface ExternalConfiguration {
+  [key: string]: External;
```

Comment:
According to config.json schema you can define external in two ways: either as an object, which you've implemented, or as a string that points directly to the script's URL, which we're missing.

Reply:
fixed